lcc_comchain_base/models/company.py

```
@@ -19,3 +26,54 @@ def _default_messages(self):
+            for block_nb in range(company_id.last_block_checked_nb + 1, last_block_id + 1):
+                _logger.info("Get Debit Transactions - read block %s" % block_nb)
+                # Get the block transactions
+                block_txs = pyc3l.BlockByNumber(block_nb).bc_txs
+                for tx in block_txs:
+                    if not tx.currency:
+                        continue # this transaction is not linked with a currency
+                    if not tx.currency.name.lower() == company_id.comchain_currency_name.lower():
+                        continue # this transaction is about another smart contract
+                    full_tx = tx.full_tx
+                    if not full_tx.is_cc_transaction:
+                        continue # this transaction is not a digital currency transfer
+                    if not full_tx.direction == 1:
+                        continue # this transaction is not a reconversion
+                    _logger.info("Transactions %s retrieved" % tx.hash)
+                    sender_adress = full_tx.add2.lstrip("0x")
+                    matched_wallet = self.env["res.partner.backend"].search([("comchain_id", "ilike", sender_adress)])
+                    if len(matched_wallet) == 0:
+                        continue # this transactions comes from a wallet no registered in current Odoo database
+                    string_amount = str(full_tx.sent)
+                    res.append({
+                        "sender": "comchain:%s" % sender_adress,
+                        "amount": f"{string_amount[0:-2]}.{string_amount[-2:]}",
+                        "tx_id": tx.hash,
+                        "tx_timestamp": datetime.fromtimestamp(int(full_tx.receivedat["value"])),
+                    })
+
+            company_id.last_block_checked_nb = last_block_id
+        except Exception as e:
+            _logger.error(tools.format_last_exception())
+
+        return res
```

res might be incompletely filled with some transaction of the current block. You should probably reraise the exception ? You probably have a good reason why you didn't.

---

fixed: we use a tmp list `comchain_res` to fill the retrieved transaction, and this `comchain_res` is added to `res` only with all the process went smoothly.